src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java

```diff
@@ -261,6 +267,53 @@ public void should_reject_file_that_is_not_a_valid_zip_archive() throws Exceptio
         errorWasSent(SAMPLE_ZIP_FILE_NAME, ErrorCode.ERR_ZIP_PROCESSING_FAILED);
     }
 
+    @Test
+    public void should_reject_file_which_has_duplicate_cdn_number() throws Exception {
+        // given
+        byte[] zipBytes = zipDir("zipcontents/ok");
+
+        uploadToBlobStorage(SAMPLE_ZIP_FILE_NAME, zipBytes);
+
+        // and
+        Pdf pdf = new Pdf("1111002.pdf", toByteArray(getResource("zipcontents/ok/1111002.pdf")));
+
+        given(documentManagementService.uploadDocuments(ImmutableList.of(pdf)))
+            .willReturn(ImmutableMap.of(
+                "1111002.pdf", DOCUMENT_URL2
+            ));
+
+        // when
+        processor.processBlobs();
+
+        // and given
+        String filenameForDuplicate = "DUP" + SAMPLE_ZIP_FILE_NAME;
+        byte[] duplicateZipBytes = zipDir("zipcontents/duplicate_cdn_with_ok");
```

Comment:
`duplicate_cdn_with_ok` is invalid because it has 2 identical dcns in the same envelope (which is going to be checked before we even try to save to DB.
We can just send the `ok` envelope under a different name.